themes/bootstrap3/templates/Recommend/SideFacets.phtml

```diff
@@ -7,238 +7,91 @@
 <? if ($results->getResultTotal() > 0): ?>
   <h4><?=$this->transEsc(isset($this->overrideSideFacetCaption) ? $this->overrideSideFacetCaption : 'Narrow Search')?></h4>
 <? endif; ?>
-<? $checkboxFilters = $results->getParams()->getCheckboxFacets(); if (count($checkboxFilters) > 0): ?>
-<?
-  $html = '';
-  $shown = 0;
-  foreach ($checkboxFilters as $current) {
-    $html .= '<label class="checkbox';
-    if($results->getResultTotal() < 1 && !$current['selected'] && !$current['alwaysVisible']) {
-      $html .= ' hidden';
-    } else {
-      $shown ++;
+<? $checkboxFilters = $results->getParams()->getCheckboxFacets(); ?>
+<? if (count($checkboxFilters) > 0):
+    $checkboxesShown = false;
+    foreach ($checkboxFilters as &$current) {
+      if ($results->getResultTotal() < 1 && !$current['selected'] && !$current['alwaysVisible']) {
+        $current['shown'] = false;
+      } else {
+        $checkboxesShown = true;
+        $current['shown'] = true;
+      }
     }
-    $html .= '"><input type="checkbox" name="filter[]" value="'.$this->escapeHtmlAttr($current['filter']).'"
-      '. ($current['selected'] ? 'checked="checked"' : '') .' id="'.$this->escapeHtmlAttr(str_replace(' ', '', $current['desc'])).'"
-      onclick="document.location.href=\''.($current['selected'] ? $results->getUrlQuery()->removeFilter($current['filter']) : $results->getUrlQuery()->addFilter($current['filter'])).'\';" />'.$this->transEsc($current['desc']).'</label>';
-  }
-?>
-  <div class="checkboxFilter<?if($shown == 0):?> hidden<? endif; ?>"><?=$html ?></div>
+  ?>
+  <div class="checkboxFilter<?if(!$checkboxesShown):?> hidden<? endif; ?>">
+    <? foreach ($checkboxFilters as $current): ?>
```

Comment:
The old code was ugly, but it avoided looping through this data twice. Is there a more efficient compromise between the two solutions?